File: redsheet-portal/src/app/database-models/v-personality.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vPersonality {
    constructor(vPersonality?: vPersonality )  {
        let properties = [ "description", "personalityColor", "personalityId", "personalityKey", "personalityKeyText",];
        for (let property of properties)
            if (vPersonality && vPersonality[property])
                this[property] = vPersonality[property];
    }
 
    @maxLength(1000)
	description : string =   undefined;
 
    @required()
    @maxLength(50)
	personalityColor : string =   undefined;
 
	personalityId : number =   0 ;
 
    @maxLength(50)
	personalityKey : string =   undefined;
 
    @maxLength(10)
	personalityKeyText : string =   undefined;


}

File: RedSheet.DbEntities/Models/vPersonality.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vPersonalities", Schema = "dbo")]
	[Serializable]
	public partial class vPersonality  
	{
 
	[MaxLength(1000)]
	public string Description { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string PersonalityColor { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int PersonalityId { get; set; } 
 
	[MaxLength(50)]
	public string PersonalityKey { get; set; } 
 
	[MaxLength(10)]
	public string PersonalityKeyText { get; set; } 

	public vPersonality()
	{
    }

    }
}

